src/vehicle.cpp

```diff
@@ -2086,7 +2092,10 @@ bool vehicle::remove_carried_vehicle( const std::vector<int> &carried_parts )
     }
 
     std::vector<point> new_mounts;
-    new_vehicle->name = veh_record.substr( vehicle_part::name_offset );
+    std::size_t pos = static_cast<int>( veh_record.find( "<turn>" ) );
+    pos = ( pos != veh_record.length() ) ? pos - 7 : pos;
+
+    new_vehicle->name = veh_record.substr( vehicle_part::name_offset, pos );
```

Comment:
veh_record is generated with the name of the vehicle, correct?
I don't know if this will cause problems, but this could probably use testing with some unicode vehicle names to ensure if behaves correctly.

Reply:
The new thing that I am saying needs testing is the string position manipulation, which is new.
You should probably split the change to add the turn to the id and the UI changes into separate PRs.

Reply:
tested successfull and discussed in discord